internals/overlord/overlord.go

```diff
@@ -249,12 +275,10 @@ func (o *Overlord) Extension() Extension {
 	return o.extension
 }
 
-func loadState(statePath string, restartHandler restart.Handler, backend state.Backend) (*state.State, *restart.RestartManager, error) {
-	timings := timing.Start("", "", map[string]string{"startup": "load-state"})
-
+func getCurrentBootID() (string, error) {
 	curBootID, err := osutil.BootID()
 	if err != nil {
-		return nil, nil, fmt.Errorf("fatal: cannot find current boot ID: %v", err)
+		return curBootID, fmt.Errorf("fatal: cannot find current boot ID: %v", err)
```

Comment:
Nit: better to explicitly return the zero value `""` in the error case (also below).